agent/src/trident.rs

```diff
@@ -264,9 +264,9 @@ impl Trident {
             }
         };
 
-        let hostname = match config.override_os_hostname.as_ref() {
+        let hostname = match &config.override_os_hostname {
             Some(name) => name.to_owned(),
-            None => get_hostname().unwrap_or_default(),
+            None => get_hostname().unwrap_or("unknown-host".to_owned()),
```

Comment:
We want to have empty string here because a valid hostname can be "unknown-host".

Reply:
Well, this change has been reverted.